### PR TITLE
Add hyphenopoly via single package.json w/ npm auto-update 

### PR DESCRIPTION
**Pull request by jefferyto**
# Pull request template for new library

Pull request for issue: #
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

## Profile of the lib

* Git repository (required): https://github.com/mnater/Hyphenopoly
* Official website (optional, not the repository): https://mnater.github.io/Hyphenopoly/
* NPM package url (optional): https://www.npmjs.com/package/hyphenopoly
* License and its reference: MIT https://github.com/mnater/Hyphenopoly/blob/master/LICENSE
* GitHub / Bitbucket popularity (required):
  * Count of watchers: 17
  * Count of stars: 250
  * Count of forks: 22
* NPM download stats (optional):
  * Downloads in the last day:
  * Downloads in the last week: 406
  * Downloads in the last month:

## Essential checklist

* [ ] I'm the author of this library
  * [ ] I would like to add link to the page of this library on CDNJS on website / readme
* [x] This lib was not found on cdnjs repo
* [x] Doesn't already exist / has duplicate issue and PR
* [x] The lib has notable popularity
  * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  * [x] More than 800 downloads stats per month on npm registry
* [x] Project has public repository on famous online hosting platform (or been hosted on npm)

## Auto-update checklist

* [x] Has valid tags for each versions (for git auto-update)
* [x] Auto-update setup
* [x] Auto-update target/source is valid.
* [x] Auto-update filemap is correct.

## Git commit checklist

* [x] The first line of commit message is less then 50 chars; clean, clear and easy to understand.
* [x] The parent of the commit(s) in the PR is not older than 3 days.
* [x] Pull request is sent from a non-master branch with a meaningful name.
* [x] Separate unrelated changes into different commits.
* [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
* [x] Close corresponding issue in commit message
* [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by jefferyto** [comment]
@dakshshah96 @sufuf3 @sashberd @extend1994 would you mind taking a look?

Hyphenopoly is the successor to [Hyphenator](https://github.com/mnater/Hyphenator), another JS hyphenation library (also [available in cdnjs](https://github.com/cdnjs/cdnjs/tree/master/ajax/libs/Hyphenator)). Hyphenator is not maintained anymore (the last release was in October 2017); all of the effort is going into Hyphenopoly.

**Comment by PeterDaveHello** [comment]
@jefferyto I'll take a look :)

**Comment by jefferyto** [comment]
@PeterDaveHello thanks :joy:

**Comment by jefferyto** [comment]
Any update on this?

**Comment by jefferyto** [comment]
Is there some other way I should submit this?

File: ajax/libs/hyphenopoly/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "type": "git",
+    "url": "https://github.com/mnater/Hyphenopoly"
+  },
+  "filename": "Hyphenopoly.js",
+  "npmName": "hyphenopoly",
+  "npmFileMap": [
+    {
+      "basePath": "min",
+      "files": [
+        "**/*"
```
Comment (PeterDaveHello) [inline ajax/libs/hyphenopoly/package.json]: I don't think we support file extention like `hpb` and `wasm`, are they required to this library? If so, we'd need to contact Cloudflare to evaluate of supporting them.

cc @MattIPv4

Comment (MattIPv4) [reply ajax/libs/hyphenopoly/package.json]: @PeterDaveHello `wasm` has been requested to be added but has not yet been added (see #13145 cc @dknecht) and, as far as I know, `hpb` is not an extension in the whitelist.

Comment (jefferyto) [reply ajax/libs/hyphenopoly/package.json]: From the [project readme](https://github.com/mnater/Hyphenopoly#readme), the `hpb` files contain hyphenation patterns in a binary format. `hyphenEngine.wasm` is for "creating pattern trie and finding hyphenation points". I believe both are required.

(There is also a `hyphenEngine.asm.js` for browsers that don't support wasm, but I believe the library only does a wasm feature test to determine which file to load, i.e. `hyphenEngine.asm.js` isn't a fallback if loading `hyphenEngine.wasm` failed.)

Comment (MattIPv4) [reply ajax/libs/hyphenopoly/package.json]: @simon-says could you help with getting `hpb` & `wasm` added to the whitelist?

Comment (jefferyto) [reply ajax/libs/hyphenopoly/package.json]: ping @simon-says @MattIPv4 - any progress on this?

**Comment by xtuc** [comment]
hpb has been added to the extension whitelist

**Comment by jefferyto** [comment]
When might this package appear on cdnjs.com?

**Pull request merged**